plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/IsNullPushdownDataTypeTest.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.clickhouse;
+
+import io.trino.Session;
+import io.trino.spi.type.Type;
+import io.trino.sql.planner.plan.FilterNode;
+import io.trino.sql.query.QueryAssertions;
+import io.trino.sql.query.QueryAssertions.QueryAssert;
+import io.trino.testing.MaterializedResult;
+import io.trino.testing.QueryRunner;
+import io.trino.testing.datatype.ColumnSetup;
+import io.trino.testing.datatype.DataSetup;
+import io.trino.testing.sql.TemporaryRelation;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.IntStream;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static io.trino.plugin.clickhouse.IsNullPushdownDataTypeTest.PushdownImplementation.ALWAYS;
+import static io.trino.plugin.clickhouse.IsNullPushdownDataTypeTest.PushdownImplementation.CONNECTOR_EXPRESSION_ONLY;
+import static io.trino.plugin.clickhouse.IsNullPushdownDataTypeTest.PushdownImplementation.TUPLE_DOMAIN_ONLY;
+import static io.trino.plugin.clickhouse.IsNullPushdownDataTypeTest.Scenario.IS_NOT_NULL;
+import static io.trino.plugin.clickhouse.IsNullPushdownDataTypeTest.Scenario.IS_NULL;
+import static java.lang.String.format;
+import static java.util.Collections.unmodifiableList;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.joining;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public final class IsNullPushdownDataTypeTest
+{
+    private static final int SPECIAL_COLUMNS = 1;
+
+    private final List<TestCase> testCases = new ArrayList<>();
+    private final PushdownImplementation pushdownImplementation;
+
+    private IsNullPushdownDataTypeTest(PushdownImplementation pushdownImplementation)
+    {
+        this.pushdownImplementation = requireNonNull(pushdownImplementation, "pushdownImplementation is null");
+    }
+
+    public static IsNullPushdownDataTypeTest create(PushdownImplementation pushdownImplementation)
+    {
+        return new IsNullPushdownDataTypeTest(pushdownImplementation);
+    }
+
+    public IsNullPushdownDataTypeTest addRoundTrip(String inputType, String inputLiteral, Type expectedType, String expectedLiteral)
```

Comment:
RoundTrip might not be an exact method name right ? We are focusing on the null pushdown